managed/resource_cluster_db_audit_export_config.go

```diff
@@ -0,0 +1,394 @@
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var plan DbAuditExporterConfig
+	var accountId, message string
+	var getAccountOK bool
+
+	resp.Diagnostics.Append(getClusterDbAuditLogConfigPlan(ctx, req.Plan, &plan)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the DB audit log configuration")
+		return
+	}
+
+	apiClient := r.p.client
+	accountId, getAccountOK, message = getAccountId(ctx, apiClient)
+	if !getAccountOK {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	projectId, getProjectOK, message := getProjectId(ctx, apiClient, accountId)
+	if !getProjectOK {
+		resp.Diagnostics.AddError("Unable to get project ID", message)
+		return
+	}
+
+	exporterId := plan.ExporterID.Value
+
+	_, err := GetConfigByNameorID(accountId, projectId, exporterId, "", apiClient)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to associate telemetry exporter to cluster", GetApiErrorDetails(err))
+		return
+	}
+
+	clusterId := plan.ClusterID.Value
+	_, err = GetClusterByNameorID(accountId, projectId, clusterId, "", apiClient)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to associate metrics exporter to cluster", GetApiErrorDetails(err))
+		return
+	}
+
+	dbAuditExporterConfigSpec, err := getDbAuditExporterConfigSpec(plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to obtain DbAuditExporterConfigSpec", GetApiErrorDetails(err))
+		return
+	}
+
+	response, _, err := apiClient.ClusterApi.AssociateDbAuditExporterConfig(ctx, accountId, projectId, clusterId).DbAuditExporterConfigSpec(*dbAuditExporterConfigSpec).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to associate db audit log configuration to cluster", GetApiErrorDetails(err))
+		return
+	}
+
+	err = associateDbAuditExporterConfigWithRetry(ctx, apiClient, accountId, projectId, clusterId)
+
+	if err != nil {
+		errorSummary := fmt.Sprintf("Unable to associate db audit log config to cluster: %s", clusterId)
+		resp.Diagnostics.AddError(errorSummary, "The operation timed out waiting for db audit log config cluster association.")
+		return
+	}
+
+	configId := response.Data.Info.Id
+	plan.ConfigID.Value = configId
+
+	dae, readOK, err := resourceAssociateDbAuditLogConfigClusterRead(ctx, accountId, projectId, clusterId, configId, apiClient)
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of Db Audit log config cluster association ", GetApiErrorDetails(err))
+		return
+	}
+
+	diags := resp.State.Set(ctx, &dae)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func associateDbAuditExporterConfigWithRetry(ctx context.Context, apiClient *openapiclient.APIClient, accountId, projectId, clusterId string) error {
```

Comment:
Why do you need this to be a function ? Inline should be fine.

Reply:
Added this in a separate function because I didn't want to overload the `Create` method. If the general way is to add it inline then I'll do that